packages/main/src/Menu.ts

```diff
@@ -266,22 +273,73 @@ class Menu extends UI5Element {
 		return this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
 	}
 
+	get _list() {
+		return this.shadowRoot!.querySelector<List>("[ui5-list]")!;
+	}
+
+	get _menuItemGroups() {
+		return this.items.filter((item) : item is MenuItemGroup => item.isGroup);
+	}
+
+	// Return only the menu items, excluding separators and items that belong to groups
 	get _menuItems() {
-		return this.items.filter((item): item is MenuItem => !item.isSeparator);
+		return this.items.filter((item) : item is MenuItem => !item.isSeparator && !item.isGroup);
+	}
+
+	get _menuItemsAll() {
+		const items: MenuItem[] = [];
```

Comment:
One very minor from my side as well, which I missed - please use the `Array<IMenuItem>` or `Array<MenuItem>` types syntax so we could be consistent.

Reply:
Done